src/main/java/picard/nio/GoogleStorageUtils.java

```diff
@@ -0,0 +1,46 @@
+package picard.nio;
+
+import com.google.cloud.http.HttpTransportOptions;
+import com.google.cloud.storage.StorageOptions;
+import com.google.cloud.storage.contrib.nio.CloudStorageConfiguration;
+import com.google.cloud.storage.contrib.nio.CloudStorageFileSystemProvider;
+import shaded.cloud_nio.com.google.api.gax.retrying.RetrySettings;
+import shaded.cloud_nio.org.threeten.bp.Duration;
+
+
+/**
+ * Created by farjoun on 11/13/17.
+ */
+class GoogleStorageUtils {
+
+    public static void initialize() {
+        CloudStorageFileSystemProvider.setDefaultCloudStorageConfiguration(GoogleStorageUtils.getCloudStorageConfiguration(20));
+        CloudStorageFileSystemProvider.setStorageOptions(GoogleStorageUtils.setGenerousTimeouts(StorageOptions.newBuilder()).build());
+    }
+
+    /** The config we want to use. **/
+    private static CloudStorageConfiguration getCloudStorageConfiguration(int maxReopens) {
+        return CloudStorageConfiguration.builder()
+                // if the channel errors out, re-open up to this many times
+                .maxChannelReopens(maxReopens)
+                .build();
+    }
+
+    private static StorageOptions.Builder setGenerousTimeouts(StorageOptions.Builder builder) {
+        return builder
+                .setTransportOptions(HttpTransportOptions.newBuilder()
+                        .setConnectTimeout(120_000)
+                        .setReadTimeout(120_000)
+                        .build())
+                .setRetrySettings(RetrySettings.newBuilder()
+                        .setMaxAttempts(15)
+                        .setMaxRetryDelay(Duration.ofMillis(256_000L))
+                        .setTotalTimeout(Duration.ofMillis(4000_000L))
+                        .setInitialRetryDelay(Duration.ofMillis(1000L))
+                        .setRetryDelayMultiplier(2.0)
+                        .setInitialRpcTimeout(Duration.ofMillis(180_000L))
+                        .setRpcTimeoutMultiplier(1.0)
+                        .setMaxRpcTimeout(Duration.ofMillis(180_000L))
+                        .build());
```

Comment:
There are a A LOT of settings here, do you want to have them able to be set a la Defaults.java (or GoogleStorageDefaults.java)?

Reply:
I think that eventually we will want to expose these...but as there's going to be a mechanism for exposing these parameters in GATK, where NIO is dealt with in a much better way, I'd like to consider options that make configuring picard and GATK easier...so I'd prefer to wait a little (especially given how few tools currently support NIO in picard). Is that reasonable?